models/survey/src/types.ts

```diff
@@ -13,6 +13,162 @@
+//  @Model(survey.class.Survey, core.class.Space, DOMAIN_SURVEY)
+/**
+ * @public
+ */
+@Model(survey.class.Survey, core.class.Space)
+@UX(survey.string.Survey, survey.icon.Survey)
+export class TSurvey extends TSpace implements Survey {
+  @Prop(TypeString(), survey.string.SurveyName, { defaultValue: '' })
+  @Index(IndexKind.FullText)
+  declare name: string
+
+  @Prop(TypeString(), core.string.Description, { defaultValue: '' })
+  @Hidden()
+  declare description: Markup
+
+  @Prop(TypeBoolean(), core.string.Archived, { defaultValue: false })
+  @Hidden()
+  declare archived: boolean
+
+  @Prop(TypeBoolean(), core.string.Private, { defaultValue: true })
+  @Hidden()
+  declare private: boolean
+
+  @Prop(Collection(survey.class.Question), survey.string.Questions, { defaultValue: 0 })
+    questions: CollectionSize<Question> = 0
+}
+
+/**
+ * @public
+ */
+@Model(survey.class.Question, core.class.AttachedDoc, DOMAIN_SURVEY)
+export class TQuestion<Data extends QuestionData> extends TAttachedDoc implements Question<Data> {
+  @Prop(TypeRef(survey.class.Survey), core.string.AttachedTo)
+  declare attachedTo: Ref<Survey>
+
+  @Prop(TypeRank(), survey.string.Rank)
+  @Hidden()
+    rank!: Rank
+
+  @Prop(TypeQuestionData(), survey.string.QuestionData)
+    data!: Data
+}
+
+/**
+ * @public
+ */
+@Model(survey.class.QuestionData, core.class.Obj)
+export class TQuestionData extends TObj implements QuestionData {}
+
+export function TypeQuestionData (): Type<QuestionData> {
+  return { _class: survey.class.QuestionData, label: survey.string.QuestionData }
+}
+
+/**
+ * @public
+ */
+export function TypeRadioButtonsOption (): Type<RadioButtonsOption> {
+  return { _class: survey.class.RadioButtonsOption, label: survey.string.Option }
+}
+
+/**
+ * @public
+ */
+@Model(survey.class.RadioButtons, survey.class.QuestionData)
+@UX(survey.string.RadioButtons, survey.icon.RadioButtons)
+export class TRadioButtons extends TQuestionData implements RadioButtons {
+  @Prop(TypeString(), survey.string.QuestionText, { defaultValue: '' })
+    text: string = ''
+
+  @Prop(ArrOf(TypeRadioButtonsOption()), survey.string.Options, { defaultValue: [] })
+    options: RadioButtonsOption[] = []
+}
+
+/**
+ * @public
+ */
+export function TypeCheckboxesOption (): Type<CheckboxesOption> {
+  return { _class: survey.class.CheckboxesOption, label: survey.string.Option }
+}
+
+/**
+ * @public
+ */
+@Model(survey.class.Checkboxes, survey.class.QuestionData)
+@UX(survey.string.Checkboxes, survey.icon.Checkboxes)
+export class TCheckboxes extends TQuestionData implements Checkboxes {
+  @Prop(TypeString(), survey.string.QuestionText, { defaultValue: '' })
+    text: string = ''
+
+  @Prop(ArrOf(TypeCheckboxesOption()), survey.string.Options, { defaultValue: [] })
+    options: RadioButtonsOption[] = []
+}
+
+/**
+ * @public
+ */
+@Model(survey.class.Info, survey.class.QuestionData)
+export class TInfo extends TQuestionData implements Info {
```

Comment:
Not supported in UI yet. I need this type as an example of non-interactive and non-assessible question data type